task: Spring Boot Security: add User entity. Executor: Oleksii Bashkirov.

OleksiiBashkirov opened the pull request:
feat: Implement user registration with validation and exception handling
- Created User entity with fields: id, email, password, firstName, lastName, shippingAddress.
- Added necessary DTOs (UserRegistrationRequestDto, UserResponseDto) with validation annotations.
- Implemented FieldMatch annotation to ensure password and repeatPassword fields match.
- Created a mapper for converting between User and DTOs.
- Implemented AuthenticationController with a registration endpoint (POST: /api/auth/registration).
- Added to CustomGlobalExceptionHandler to handle RegistrationException and provide meaningful error responses.
- Configured database migrations for creating and populating the users and books tables.
- Tested the registration endpoint with various cases using Postman to ensure proper validation and error handling.

OleksiiBashkirov committed:
feat: Implement user registration with validation and exception handling

- Created User entity with fields: id, email, password, firstName, lastName, shippingAddress.
- Added necessary DTOs (UserRegistrationRequestDto, UserResponseDto) with validation annotations.
- Implemented FieldMatch annotation to ensure password and repeatPassword fields match.
- Created a mapper for converting between User and DTOs.
- Implemented AuthenticationController with a registration endpoint (POST: /api/auth/registration).
- Added to CustomGlobalExceptionHandler to handle RegistrationException and provide meaningful error responses.
- Configured database migrations for creating and populating the users and books tables.
- Tested the registration endpoint with various cases using Postman to ensure proper validation and error handling.

OleksiiBashkirov committed:
- removed the extra line.

Elena-Bruyako commented on src/main/java/mate/academy/annotation/FieldMatch.java:
```diff
@@ -0,0 +1,25 @@
+package mate.academy.annotation;
```
```suggestion
package mate.academy.validation;
```

Elena-Bruyako commented on src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -29,4 +29,9 @@ public ResponseEntity<String> handleGenericException(Exception ex, WebRequest re
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<String> handleRegistrationException(RegistrationException ex) {
+        return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
```
```suggestion
        return new ResponseEntity<>(ex.getMessage(), HttpStatus.CONFLICT);
```

Elena-Bruyako commented on src/main/java/mate/academy/mapper/UserMapper.java:
```diff
@@ -0,0 +1,15 @@
+public interface UserMapper {
+    User toModel(UserRegistrationRequestDto requestDto);
+
+    @Mapping(source = "shippingAddress", target = "shippingAddress")
```
Do you really need this?

Elena-Bruyako commented on src/main/java/mate/academy/model/User.java:
```diff
@@ -0,0 +1,33 @@
+import lombok.Data;
+
+@Entity
+@Data
```
Don't use `@Data` for entities

Elena-Bruyako commented on src/main/java/mate/academy/model/User.java:
```diff
@@ -0,0 +1,33 @@
+    @Column(nullable = false)
+    private String lastName;
+
+    @Column(name = "shipping_address")
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+        }
+
+        User user = userMapper.toModel(requestDto);
+        User savedUser = userRepository.save(user);
+        return userMapper.toDto(savedUser);
```
```suggestion
        userRepository.save(user);
        return userMapper.toDto(user);
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Email already in use");
```
add email to exception message

Elena-Bruyako reviewed (changes requested):
Good job. Let's improve it

OleksiiBashkirov committed:
- Refactored code:
  - Moved FieldMatch annotation to validation package.
  - Changed HTTP status for RegistrationException to CONFLICT.
  - Removed unnecessary @Mapping in UserMapper.
  - Used @Getter and @Setter instead of @Data for User entity.
  - Improved exception message in UserServiceImpl.
